src/ucp/core/ucp_worker.c

```diff
@@ -2731,3 +2696,60 @@ void ucp_worker_discard_uct_ep(ucp_worker_h worker, uct_ep_h uct_ep,
 
     ucp_worker_discard_tl_uct_ep(worker, uct_ep, ep_flush_flags);
 }
+
+static int ucp_worker_discard_remove_filter(const ucs_callbackq_elem_t *elem,
+                                            void *arg)
+{
+    return (elem->cb == ucp_worker_discard_uct_ep_progress) ||
+           (elem->cb == ucp_worker_discard_uct_ep_destroy_progress);
+}
+
+void ucp_worker_destroy(ucp_worker_h worker)
```

Comment:
why need to move ucp_worker_destroy function?

Reply:
to avoid defining forward declaration for `ucp_worker_discard_remove_filter`

Reply:
since ucp_worker_discard_remove_filter is new function, we can just define it before..

Reply:
done